app/src/main/java/com/amaze/filemanager/filesystem/FileUtil.java

```diff
@@ -655,21 +655,26 @@ public static boolean mktextfile(String data, String path, String fileName) {
     FileOutputStream out = null;
     OutputStreamWriter outputWriter = null;
     try {
-      f.createNewFile();
-      out = new FileOutputStream(f, false);
-      outputWriter = new OutputStreamWriter(out);
-      outputWriter.write(data);
-      return true;
+      if (f.createNewFile()) {
+        out = new FileOutputStream(f, false);
+        outputWriter = new OutputStreamWriter(out);
+        outputWriter.write(data);
+        return true;
+      } else {
+        return false;
```

Comment:
Are we sure we want to use the "finally executes before return"?

Reply:
Any implications on using this? If we're to prevent leaks this seems a better way to go.